indico/modules/events/registration/controllers/display.py

```diff
@@ -126,6 +126,9 @@ def _is_checkin_visible(reg):
         return reg.registration_form.publish_checkin_enabled and reg.checked_in
 
     def _merged_participant_list_table(self):
+        def _is_participant(reg):
+            return reg.event.is_user_registered(session.user, reg.registration_form)
```

Comment:
I would put this logic inside `Registration.is_publishable` itself, and pass it the regform instead of the `is_participant` bool.

Also, if you search for uses of `is_publishable`, you're actually missing this check in lots of places, such as the participant lists for meetings/lectures and the hidden participant count...

Reply:
Well, it would be cumbersome to do that as it means we need the user context every time calling is_publishable.
Also the 'is_participant' (for the registration form) depends on the context:
Yes, in the context of displaying Participant List
No, in context of "get_published_registrations" (Get a list of published registrations for an event) or "count_hidden_registrations" (the count is for the whole event). 

So we should keep as it is:
the registration **is_publishable or not** will depend on the context if the user can be considered as a participant or non...
(Fyi, the other calls of is_publishable is only in the test, and in get_published_registrations, count_hidden_registrations).

Reply:
Well, to avoid the context thing you can pass both the regform and the user to `is_publishable`

Either way, this check needs to be applied in `get_published_registrations`, because it is used to display the registrations in the event header (in case of meetings and lectures) and in the timetable for conferences if you select an "Indico" theme for it. Same applies to `count_hidden_registrations`, or the count will turn out wrong. I tested it and it still shows registrations that you're not supposed to see

Reply:
The current get_published_registrations and count_hidden_registrations show always the total number of registrations, so when a user is allowed to see other participants, he still see the total number and the number of hidden,
you want to change that to display calculation based on registration form ?
<img width="578" alt="image" src="https://github.com/indico/indico/assets/11231950/3269173a-91af-4a32-8045-7b407064030b">

Reply:
that is not where those methods are used..

let me show you. right now, with your PR, this is what I see as "John Doe" on the participants list:
![image](https://github.com/indico/indico/assets/27357203/bebdb382-0a94-4da7-beee-2c7cb7665e2f)

I myself am enrolled in the "Another" form, but I'm hidden since I chose consent only to participants. This is what I see as myself:
![image](https://github.com/indico/indico/assets/27357203/2810bab8-15bc-4e65-bf5d-c62b4c55838b)

This is all fine. But as "John Doe", this is what I see on the timetable:
![image](https://github.com/indico/indico/assets/27357203/0a19425b-3173-49c8-a99b-8058d88a6186)

My name shouldn't be visible there. Also, since my name wouldn't show up, the `count_hidden_registrations` would have to return 5 instead of 4